File: backend/src/main/java/com/ll/hotel/domain/hotel/option/entity/HotelOption.java
package com.ll.hotel.domain.hotel.option.entity;

import com.ll.hotel.domain.hotel.hotel.entity.Hotel;
import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "hotel_option")
public class HotelOption extends BaseOption {
    @ManyToMany(mappedBy = "hotelOptions")
    private Set<Hotel> hotels = new HashSet<>();

    @Builder
    public HotelOption(String name) {
        super(name);
        this.hotels = (hotels != null) ? hotels : new HashSet<>();
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/option/entity/HotelOption.kt
package com.ll.hotel.domain.hotel.option.entity

import com.ll.hotel.domain.hotel.hotel.entity.Hotel
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.ManyToMany

@Entity
class HotelOption(
    name: String,

    @ManyToMany(mappedBy = "hotelOptions", fetch = FetchType.LAZY)
    val hotels: MutableSet<Hotel> = mutableSetOf()

) : BaseOption(name)